vector/src/main/java/im/vector/app/features/spaces/explore/SpaceDirectoryController.kt

```diff
@@ -194,4 +208,20 @@ class SpaceDirectoryController @Inject constructor(
             }
         }
     }
+
+    class RoomSearchMatchFilter : Predicate<SpaceChildInfo> {
+        var filter: String = ""
+
+        override fun test(spaceChildInfo: SpaceChildInfo): Boolean {
+            if (filter.isEmpty()) {
+                // No filter
+                return true
+            }
+            // if filter is "Jo Do", it should match "John Doe"
+            return filter.split(" ").all {
+                spaceChildInfo.name?.contains(it, ignoreCase = true).orFalse() ||
+                        spaceChildInfo.topic?.contains(it, ignoreCase = true).orFalse()
+            }
```

Comment:
I think it's ok to not take ordering in account since there should not be so many items after filter is applied. Also that's actually a copy-paste and this is 4th copy of this code, so I think I'll create a small refactoring issue for this later.

Reply:
I've just found that there is exactly same implementation of SpaceChild MatchFilter, so I've reused that instead of copy paste. 

And about issue - I'm not sure if it's actually could be refactored, so I decided to take a look on that after this will be merged and decide what to do

Reply:
IMO it should be aligned with the other platforms to have the same results, so I agree it would be the object of a dedicated PR.
The same issue exists about the search in the user directory which leads to different results depending on the platform.